testsuite/integration/manualmode/src/test/java/org/wildfly/test/manual/elytron/seccontext/DynamicSSLContextTestCase.java

```diff
@@ -0,0 +1,350 @@
+        }
+        ManagementClient managementClient = null;
+        try {
+            managementClient = new ManagementClient(TestSuiteEnvironment.getModelControllerClient(), TestSuiteEnvironment.getServerAddress(), 9990, "remote+http");
+
+            addHTTPSListener(managementClient, KEYSTORES_DIR, SERVER_KEYSTORE1_FILENAME, SERVER_KEY_STORE1, SERVER_KEY_MANAGER1, SERVER_TRUSTSTORE1_FILENAME, SERVER_TRUST_STORE1, SERVER_TRUST_MANAGER1, SERVER_SSL_CONTEXT1, HTTP_LISTENER_1, SOCKET_BINDING_1, 10443);
+            addHTTPSListener(managementClient, KEYSTORES2_DIR, SERVER_KEYSTORE1_FILENAME, SERVER_KEY_STORE2, SERVER_KEY_MANAGER2, SERVER_TRUSTSTORE1_FILENAME, SERVER_TRUST_STORE2, SERVER_TRUST_MANAGER2, SERVER_SSL_CONTEXT2, HTTP_LISTENER_2, SOCKET_BINDING_2, 11443);
+
+            addSSLContext(managementClient, KEYSTORES_DIR, CLIENT_KEYSTORE1_FILENAME, CLIENT_KEY_STORE1, CLIENT_KEY_MANAGER1, CLIENT_TRUSTSTORE1_FILENAME, CLIENT_TRUST_STORE1, CLIENT_TRUST_MANAGER1, CLIENT_SSL_CONTEXT1);
+            addSSLContext(managementClient, KEYSTORES2_DIR, CLIENT_KEYSTORE1_FILENAME, CLIENT_KEY_STORE2, CLIENT_KEY_MANAGER2, CLIENT_TRUSTSTORE1_FILENAME, CLIENT_TRUST_STORE2, CLIENT_TRUST_MANAGER2, CLIENT_SSL_CONTEXT2);
+
+            addAuthenticationContext(managementClient);
+            configureDefaultSSLContext(managementClient);
+
+            reloadServer(managementClient);
+            waitFoServerToStart(5000, TestSuiteEnvironment.getServerAddress(), 9990);
+            deployer.deploy(DEPLOYMENT);
+        } finally {
+            IoUtils.safeClose(managementClient);
+        }
+    }
+
+    @Test
+    @RunAsClient
+    @InSequence(1)
+    public void startAndConfigureContainer() throws IOException {
+        startContainer();
+    }
+
+    @Test
+    @InSequence(2)
+    @OperateOnDeployment(DEPLOYMENT)
+    public void test() throws Exception {
+        // default ssl context is configured to be dynamic in the server
+        SSLSocketFactory sf = SSLContext.getDefault().getSocketFactory();
+        // both handshakes will succeed even though these ports require different truststore and keystore
+        SSLSocket socket = (SSLSocket) sf.createSocket("127.0.0.1", 10443);
+        socket.startHandshake();
+        socket = (SSLSocket) sf.createSocket("127.0.0.1", 11443);
+        socket.startHandshake();
+    }
+
+    @Test
+    @RunAsClient
+    @InSequence(3)
+    public void cleanContainer() throws IOException {
+        stopContainer();
+    }
+
+    private void stopContainer() throws IOException {
+        ManagementClient managementClient = null;
+        try {
+            managementClient = new ManagementClient(TestSuiteEnvironment.getModelControllerClient(), TestSuiteEnvironment.getServerAddress(), 9990, "remote+http");
+            restoreConfiguration(managementClient);
+            deployer.undeploy(DEPLOYMENT);
+        } finally {
+            IoUtils.safeClose(managementClient);
+        }
```

Comment:
This doesn't actually stop the container. :)

Reply:
Oops you're right. Fixed it